src/jit/gentree.h

```diff
@@ -4823,26 +4879,38 @@ inline GenTreePtr GenTree::MoveNext()
 //    true:      the GenTree node is accepted as a valid argument
 //    false:     the GenTree node is not accepted as a valid argumeny
 //
-inline bool GenTree::IsListForMultiRegArg()
+// Notes:
+//    For targets that don't support arguments as a list of fields, we do not support GT_FIELD_LIST.
+//
+//    Currently for AMD64 UNIX we allow a limited case where a GT_FIELD_LIST is
+//    allowed but every element must be a GT_LCL_FLD.
+//
+//    For the future targets that allow for Multireg args (and this includes the current ARM64 target),
+//    or that allow for passing promoted structs, we allow a GT_FIELD_LIST of arbitrary nodes.
+//    These would typically start out as GT_LCL_VARs or GT_LCL_FLDS or GT_INDs,
+//    but could be changed into constants or GT_COMMA trees by the later
+//    optimization phases.
+
+inline bool GenTree::IsValidCallArgument()
 {
-    if (!IsList())
+    if (OperIsList())
     {
-        // We don't have a GT_LIST, so just return true.
-        return true;
+        // GT_FIELD_LIST is the only list allowed.
+        return false;
     }
-    else // We do have a GT_LIST
+    if (OperIsFieldList())
     {
-#if defined(LEGACY_BACKEND) || !FEATURE_MULTIREG_ARGS
+#if !FEATURE_MULTIREG_ARGS
```

Comment:
Btw, FEATURE_MULTIREG_ARG is true on _TARGET_ARM_ (i.e. arm32).  That means this code will be enabled for arm32 legacy backend as well.